test-case/check-signal-stop-start.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+##
+## Case Name: check-stop-start
+## Preconditions:
+##    N/A
+## Description:
+##    Run aplay/arecord on each pipeline and use SIGSTOP and SIGCONT to
+##    simulate keyboard inputs ctrl+z and fg.
+##    kill SIGSTOP $pid will stop the thread
+##    kill SIGCONT $pid will start the thread
+## Case step:
+##    1. aplay/arecord on PCM
+##    2. use send SIGSTOP/SIGCONT to stop/start
+## Expect result:
+##    no error happen for aplay/arecord
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['i']='sleep'    OPT_DESC_lst['i']='interval time for stop/start'
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']=0.5
+
+OPT_OPT_lst['d']='duration' OPT_DESC_lst['d']='duration time'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=10
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+tplg=${OPT_VALUE_lst['t']}
+count=${OPT_VALUE_lst['c']}
+interval=${OPT_VALUE_lst['i']}
+duration=${OPT_VALUE_lst['d']}
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+declare -A CMD FILE
+CMD['playback,both']='aplay'
+FILE['playback,both']='/dev/zero'
+CMD['capture,both']='arecord'
+FILE['capture,both']='/dev/null'
+
+func_stop_start_pipeline()
+{
+    count=1
+    while(true)
+    do
+        ps -ef |grep "$pid" |grep -v grep
```

Comment:
can we use some other check logic? like `kill -0 $pid`?
@Bin-QA any suggestion?

Reply:
@Bin-QA  good suggestion. Updated.